Code/GraphMol/Wrap/MolOps.cpp

```diff
@@ -741,6 +741,40 @@ ROMol *adjustQueryPropertiesHelper(const ROMol &mol, python::object pyparams) {
   return MolOps::adjustQueryProperties(mol, &params);
 }
 
+ROMol *replaceCoreHelper(const ROMol &mol,
+                         const ROMol &core,
+                         python::object match,
+                         bool replaceDummies,
+                         bool labelByIndex,
+                         bool requireDummyMatch=false) {
+  // convert input to MatchVect
+  MatchVectType matchVect;
+
+  unsigned int length = python::extract<unsigned int>(match.attr("__len__")());
+  
+  for (unsigned int i = 0; i < length; ++i) {
+    int sz = 1;
+    if(PyObject_HasAttrString(static_cast<python::object>(match[i]).ptr(), "__len__")) {
+      sz = python::extract<unsigned int>(match[i].attr("__len__")());
+    }
+    
+    int v1,v2;
+    PRECONDITION(sz == 1 || sz == 2, "Input not a vector of (core_atom_idx,molecule_atom_idx) entries");
```

Comment:
Again, this (and the one at line 764) should throw ValueError instead of using invariants